lucille-core/src/main/java/com/kmwllc/lucille/doclet/JsonDoclet.java

```diff
@@ -0,0 +1,177 @@
+                        if (exec.getKind() != ElementKind.METHOD) continue;
+                        List<String> paramTypes = exec.getParameters().stream()
+                                .map(p -> p.asType().toString())
+                                .collect(Collectors.toList());
+                        List<String> paramNames = exec.getParameters().stream()
+                                .map(p -> p.getSimpleName().toString())
+                                .collect(Collectors.toList());
+                        Method method = null;
+                        try {
+                            method = runtimeClass.getDeclaredMethod(exec.getSimpleName().toString(),
+                                    exec.getParameters().stream().map(p -> {
+                                        try {
+                                            return Class.forName(p.asType().toString());
+                                        } catch (Exception ex) {
+                                            return Object.class;
+                                        }
+                                    }).toArray(Class[]::new));
+                        } catch (Exception ex) {
+                            continue;
+                        }
+                        MethodDescriptor md = new MethodDescriptor(method, paramNames);
+                        md.setDescription(env.getElementUtils().getDocComment(exec));
+                        methodDescriptors.add(md);
+                    }
+                }
+                // Use a constructor or reflection to set the methods field directly
+                ClassDescriptor cd = new ClassDescriptor(runtimeClass, methodDescriptors);
+                cd.setDescription(env.getElementUtils().getDocComment(type));
+                descriptors.add(cd);
+            } catch (Exception ex) {
+                reporter.print(Diagnostic.Kind.WARNING, "Failed to process "
+                    + e + ": " + ex.getMessage());
+            }
+        }
+
+        try {
+            javax.tools.JavaFileManager fileManager = javax.tools.ToolProvider.getSystemJavaCompiler().getStandardFileManager(null, null, null);
+            FileObject file = fileManager.getFileForOutput(StandardLocation.CLASS_OUTPUT, "", "javadocs.json", null);
+            try (Writer w = file.openWriter()) {
+                Gson gson = new GsonBuilder().setPrettyPrinting().create();
+                gson.toJson(descriptors, new TypeToken<List<ClassDescriptor>>(){}.getType(), w);
+            }
+        } catch (Exception e) {
+            reporter.print(Diagnostic.Kind.ERROR, "Error writing javadocs.json: " + e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Main method to run JsonDoclet as a standalone application.
+     * Uses sensible defaults for source path, package, and output file.
+     */
+    public static void main(String[] args) throws Exception {
```

Comment:
we should probably move this to a unit test instead of having a main method here.

Reply:
Not sure if deriving your own doclet requires a main - the javadoc tool requires it .. I'll dig into it